src/zoo/owl_zoo_dir.ml

```diff
@@ -3,55 +3,101 @@
  * Copyright (c) 2016-2018 Liang Wang <liang.wang@cl.cam.ac.uk>
  *)
 
+let dir = Owl_zoo_log.dir
 
-let rec _extract_zoo_gist f added =
+let parse_gist_string gist =
```

Comment:
a suggestion,  i think this function can be better written with pattern matching 

```ocaml
match s with
| gist :: [] -> ...
| gist :: version :: [] -> ...
| _ -> ....
```

Reply:
Also, because this is your core function, you should put expand rule here, at least briefly. For complicated functions, you need to have necessary inline doc.

Reply:
Agree. Documentation is surely required for it to be used by your doc and other users.